test/handler/pool_test.exs

```diff
@@ -187,16 +187,43 @@ defmodule Handler.PoolTest do
            end)
   end
 
+  test "kills a job by its given task_id" do
+    task_id = "task_1234"
+
+    config = %Pool{
+      max_workers: 5,
+      max_memory_bytes: 1024 * 1024
+    }
+
+    {:ok, pool} = Pool.start_link(config)
+
+    fun = fn ->
+      :timer.sleep(60_000)
+      {:ok, self()}
+    end
+
+    opts = [max_heap_bytes: 10 * 1024, max_ms: 1_000, task_id: task_id]
+
+    Task.async(fn -> Pool.run(pool, fun, opts) end)
+
+    # ensure pool is done initilizing worker
+    :sys.get_state(pool)
+
+    assert :ok = Pool.kill(pool, task_id)
+
+    assert {:reject, "No task with given task_id in state"} = Pool.kill(pool, task_id)
+  end
+
   describe "composed pools" do
-    test "jobs successfully delgate to the root pool" do
+    test "jobs successfully delegate to the root pool" do
       {_root, composed} = setup_composed_pools()
       opts = [max_heap_bytes: 5 * 1024, max_ms: 20]
       assert {:ok, 100} = Pool.run(composed, fn -> {:ok, 10 * 10} end, opts)
     end
 
     test "resources are marked free from both pools" do
       {_root, composed} = setup_composed_pools()
-      opts = [max_heap_bytes: 5 * 1024, max_mx: 20]
+      opts = [max_heap_bytes: 5 * 1024, max_ms: 20]
```

Comment:
Nice catch! Maybe in a future PR we should do some validation on these options to reject unknown options?